hackathon.md

```diff
@@ -0,0 +1,94 @@
+*The [`lowrisc/cheriot-rtos`][] fork only exists to hold fresh patches that aren't quite ready to be upstreamed to [`microsoft/cheriot-rtos`][] but will be.*
+
+Other repositories of note:
+- [`sonata-system`][]: holds the Sonata system RTL and bootloader which come together to generate the bitstream.
+- [`sonata-rp2040`][]: The firmware that is running on the Sonata's RP2040 microcontroller.
+- [`CHERIoT-Platform/llvm-project`][]: The CHERIoT toolchain.
+- [`cheriot-audit`][]: A tool to explore and verify the relationships between CHERIoT RTOS components.
+
+For hardware documentation, see the [Sonata system book][].
+
+[`sonata-software`]: https://github.com/lowRISC/sonata-software
+[`lowrisc/cheriot-rtos`]: https://github.com/lowRISC/cheriot-rtos
+[`microsoft/cheriot-rtos`]: https://github.com/microsoft/cheriot-rtos
+[`sonata-system`]: https://github.com/lowRISC/sonata-system
+[`sonata-rp2040`]: https://github.com/newaetech/sonata-rp2040
+[`CHERIoT-Platform/llvm-project`]: https://github.com/CHERIoT-Platform/llvm-project
+[`cheriot-audit`]: https://github.com/CHERIoT-Platform/cheriot-audit
+[CHERIoT Programmer's Guide]: https://cheriot.org/book/
+[Sonata system book]: https://lowrisc.org/sonata-system/
+
+
+## The First Build and Load
+
+I'm going to assume you've run through the whole of the [`README.md`](README.md).
+In which case, you will have ran `xmake` and built the firmware images specified in [`xmake.lua`](xmake.lua), specifically by the *firmware* rules.
+Each of the firmware targets depends on *compartments*, declared with `add_deps`, and has a number of *threads*.
+For more information about these rules, see [`cheriot-rtos/README.md`](cheriot-rtos/README.md#building-firmware-images).
+
+The only thing this repository adds is a little post-link function, `convert_to_utf2`, that converts the ELF of the firmware image into a UF2 file which can be easily loaded onto the Sonata board.
+To do this, just copy the built UF2 to the `SONATA` drive which should have mounted when you plugged the board in.
+After doing this, you should see some flashing LEDs!
+
+```sh
+xmake
+cp build/cheriot/cheriot/release/sonata_simple_demo.uf2 /run/media/hugom/SONATA/
+sync
+```
+
+To see the UART console logs, attach to `/dev/ttyUSB2` at a Baud of 115200 with your favourite terminal.
+
+```sh
+picocom /dev/ttyUSB2 -b 115200 --imap lfcrlf
+```
+
+You should see:
+
+```
+bootloader: Loading software from flash...
+bootloader: Booting into program, hopefully.
+Led Walk Raw: Look pretty LEDs!
+```
+
+If you'd like get debug messages from the RTOS Core compartments, you can enable them with the following configuration flags.
+
+```sh
+xmake config --board=sonata --debug-scheduler=y \
+    --debug-locks=y --debug-cxxrt=y \
+    --debug-token_library=y --debug-allocator=y
+xmake build
+```
+
+You may need to run `rm -rf build/ .xmake/` to remove the debug messages from the configuration.
+
```

Comment:
```suggestion
## Fallback Firmware load

We have been experiencing some reliability issues with copying firmware UF2 files to the SONATA virtual USB drive. You may find repeating the copy is sufficient to get things working but if you are having trouble we have a temporary backup load feature in the RP2040 firmware that uses the RP2040 USB boot/flash loader.

You will need to create a new UF2 from your firmware UF2 with the script in [`scripts/rp2040_uf2/`](scripts/rp2040_uf2/) 

# Script must be executed from the directory it lives in
cd scripts/rp2040_uf2
# Creates a file build/cheriot/cheriot/release/sonata_simple_demo_rp2040.uf2
./make_rp2040_uf2.sh ../../build/cheriot/cheriot/release/sonata_simple_demo.uf2

To use the new UF2 restart your board in RP2040 USB mode (hold the 'RP2040 boot' button whilst power cycling the board). Copy the new UF2 (build/cheriot/cheriot/release/sonata_simple_demo_rp2040.uf2 in the example above) to the RP2040 drive and this will load the Sonata firmware
```

(The shell stuff needs to be enclosed in backticks but these break the github UI suggestion thing)